### PR TITLE
merge 0.7.1

### PR DESCRIPTION
**Pull request by ccheetham**
Merge dev for 0.7.1 release

### CODE REVIEW

**Comment by ccheetham** [comment]
The bulk of the changes are contained in 3cc28a48e1df3f5bc7b2b8c2ca63febf0992e37a

Functionally, these changes are a reflection of the endgoal described in the doc 
[Steeltoe Tooling CLI](https://docs.google.com/document/d/16uVD7nKEzr6RLdEFHBWhnxx83i04nxodRGRnsSow8gE).  

The functional subset actually implemented is described on the [Steeltoe docs dev site](https://dev.steeltoe.io/docs/3/developer-tools/cli).

Highlights:
* organized directory structure for MVC
* renamed Execute interface to Controller to reflect MVC semantics
* removed old command impls
* added impls for `show`, `run`, `stop`
* simplified the backend since we're only supporting running locally in Docker
* similarly, state machine also simplified
* impl'd template engine

File: src/Steeltoe.Tooling/steeltoe.rc/registry/dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - Npgsql
+    - Npgsql.EntityFrameworkCore.PostgreSQL
+- type: rabbitmq
+  image: steeltoeoss/rabbitmq:3.7
+  port: 5672
+  nugets:
+    - RabbitMQ.Client
+- type: redis
+  image: steeltoeoss/redis-amd64-linux:4.0.11
+  port: 5672
```
Comment (TimHess) [inline src/Steeltoe.Tooling/steeltoe.rc/registry/dependencies.yml]:
```suggestion
  port: 6379
```

File: src/Steeltoe.Tooling/steeltoe.rc/registry/dotnet.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+netcoreapp2.1: mcr.microsoft.com/dotnet/core/sdk:2.1
+netcoreapp2.2: mcr.microsoft.com/dotnet/core/sdk:2.2
```
Comment (TimHess) [inline src/Steeltoe.Tooling/steeltoe.rc/registry/dotnet.yml:2]: Should 2.2 be here?

Comment (ccheetham) [reply src/Steeltoe.Tooling/steeltoe.rc/registry/dotnet.yml:2]: No.  I'll open a separate ticket and resolve post 0.7.1

**Review by TimHess — APPROVED** [review]

**Pull request merged**